### PR TITLE
Corey Carter

### PR DESCRIPTION
**Pull request by crc2231**

### CODE REVIEW

File: lib/author.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def name
+    @first_name + " " + @last_name
+  end
+
+  def write(book_name, book_publication_date)
+    details = {}
+    details[:author_first_name] = self.first_name
+    details[:author_last_name] = self.last_name
```
Comment (mikedao) [inline lib/author.rb:19]: I think it'd be simpler to just do first_name or last_name or just even the instance variables.

File: lib/author.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    details[:title] = book_name
+    details[:publication_date] = book_publication_date
+    book = Book.new(details)
+    books << book
```
Comment (mikedao) [inline lib/author.rb:23]: You can just do a books << Book.new(details) then just return the last book in the collection

File: lib/library.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @rental_count = {}
```
Comment (mikedao) [inline lib/library.rb:11]: I dont know if I'd store this information here, I might keep it with the book.

File: lib/library.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  def add_author(author)
+    @authors << author
+    author.books.each do |book|
+      @books << book
+      @rental_count[book] = 0
+    end
+  end
+
+  def publication_time_frame_for(author)
```
Comment (timomitchel) [inline lib/library.rb:22]: Nice job, after the updates to the spec, you got this method and all others. Great work on this challenge.

**Pull request closed without merging**